### PR TITLE
support tidb txn mode option

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
close https://github.com/pingcap/tidb-binlog/issues/981

### What is changed and how it works?
Set `tidb_txn_mode` to `optimisstic` as default to improve effiency.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### Release note
- support tidb_txn_mode for drainer dsync and use `optimistic` as default.

### CODE REVIEW

File: pkg/loader/util.go [hunk 1]
```diff
@@ -151,13 +150,16 @@ func CreateDBWithSQLMode(user string, password string, host string, port int, tl
...
+	}
 
-	return createDBWitSessions(dsn)
+	return createDBWitSessions(dsn, params)
 }
 
 // CreateDB return sql.DB
 func CreateDB(user string, password string, host string, port int, tls *tls.Config) (db *gosql.DB, err error) {
-	return CreateDBWithSQLMode(user, password, host, port, tls, nil)
+	return CreateDBWithSQLMode(user, password, host, port, tls, nil, nil)
```
Comment (july2993) [inline pkg/loader/util.go:162]: note this change the behavior of `CreateDB` which will set `allow_auto_random_explicit_insert` as `1` before.
```shell
➜  tidb-binlog git:(master) ✗ ag loader.CreateDB
tests/kafka/kafka.go
61:	sinkDB, err := loader.CreateDB("root", "", "127.0.0.1", 3306, nil)

drainer/checkpoint/mysql.go
48:var sqlOpenDB = loader.CreateDB

drainer/relay.go
46:	db, err := loader.CreateDBWithSQLMode(scfg.To.User, scfg.To.Password, scfg.To.Host, scfg.To.Port, scfg.To.TLS, scfg.StrSQLMode)

drainer/sync/mysql.go
45:var createDB = loader.CreateDBWithSQLMode

arbiter/server.go
38:	createDB  = loader.CreateDB

reparo/syncer/mysql.go
50:var createDB = loader.CreateDB
```

File: drainer/sync/util.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ type DBConfig struct {
 	Checkpoint              CheckpointConfig `toml:"checkpoint" json:"checkpoint"`
 	BinlogFileDir           string           `toml:"dir" json:"dir"`
 	BinlogFileRetentionTime int              `toml:"retention-time" json:"retention-time"`
+	TiDBTxnMode             string           `toml:"tidb-txn-mode" json:"tidb-txn-mode"`
```
Comment (july2993) [inline drainer/sync/util.go:38]: how about adding  `param`(map) for arbitrary params?
(I am ok keeping `TiDBTxnMode`)

File: pkg/loader/util.go [hunk 3]
```diff
@@ -94,14 +100,7 @@ func isUnknownSystemVariableErr(err error) bool {
 	return code == errno.ErrUnknownSystemVariable
 }
 
-func createDBWitSessions(dsn string) (db *gosql.DB, err error) {
-	// Try set this sessions if it's supported.
-	params := map[string]string{
```
[the remark this answers is not in the document]
Comment (lichunzhu) [reply pkg/loader/util.go:99]: addressed in https://github.com/pingcap/tidb-binlog/pull/1003/commits/3da00b74e5fc1b2343b9a371b73eb0079d28b3c9

File: drainer/sync/util.go [hunk 4]
```diff
@@ -29,12 +29,14 @@ type DBConfig struct {
...
-	Checkpoint              CheckpointConfig `toml:"checkpoint" json:"checkpoint"`
-	BinlogFileDir           string           `toml:"dir" json:"dir"`
-	BinlogFileRetentionTime int              `toml:"retention-time" json:"retention-time"`
+	EncryptedPassword       string            `toml:"encrypted_password" json:"encrypted_password"`
+	SyncMode                int               `toml:"sync-mode" json:"sync-mode"`
+	Port                    int               `toml:"port" json:"port"`
+	Checkpoint              CheckpointConfig  `toml:"checkpoint" json:"checkpoint"`
+	BinlogFileDir           string            `toml:"dir" json:"dir"`
+	BinlogFileRetentionTime int               `toml:"retention-time" json:"retention-time"`
+	TiDBTxnMode             string            `toml:"tidb-txn-mode" json:"tidb-txn-mode"`
```
Comment (july2993) [inline drainer/sync/util.go:38]: remove this

**Review by july2993 — COMMENTED** [review]
rest LGTM

**Review by july2993 — APPROVED** [review]
LGTM

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Comment by july2993** [comment]
/run-all-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

**Pull request merged**